src/core/QRSVG.ts

```diff
@@ -183,7 +183,8 @@ export default class QRSVG {
     const dot = new QRDot({ svg: this._element, type: options.dotsOptions.type });
 
     this._dotsClipPath = document.createElementNS("http://www.w3.org/2000/svg", "clipPath");
-    this._dotsClipPath.setAttribute("id", "clip-path-dot-color");
+    const svgId = Math.random();
```

Comment:
I don't love random values. It means you can't reference the element in CSS, and depending on how this is used, you'll get different values with each render (e.g. client and server, if applicable). Using something deterministic like a global counter might be better. The downside of that approach is that if you have _multiple instances of this library_ in your bundle, you'll get duplicate IDs. I think that's probably okay, since if that happens, you should fix that issue because it's bad on its own!

Reply:
Thanks, agreed, updated it to a global counter. I've used a Typescript `static` field which I think will work well in every case (as long as there's only one instance of the library as you said).